java/ql/src/experimental/Security/CWE/CWE-400/ThreadResourceAbuse.ql

```diff
@@ -22,7 +22,8 @@ module ThreadResourceAbuseConfig implements DataFlow::ConfigSig {
   predicate isSink(DataFlow::Node sink) { sink instanceof PauseThreadSink }
 
   predicate isAdditionalFlowStep(DataFlow::Node pred, DataFlow::Node succ) {
-    any(AdditionalValueStep r).step(pred, succ)
+    any(AdditionalValueStep r).step(pred, succ) or
```

Comment:
`AdditionalValueStep` is applied by default to all dataflow configurations, so this can actually be removed:

```suggestion
```

Reply:
Thanks for the review, @atorralba!

I'm still a bit new to taint tracking - can you clarify what makes this mix a dataflow query with a taint tracking query? Is it the use of "AdditionalValueStep" (value preserving?) alongside `TaintTracking::Global<>` and the `ThreadSourceAbuseAdditionalTaintStep`? 

I noticed the query originally added the `RunnableStartToRunStep` and `ApacheFileUploadProgressUpdateStep` classes that extended from `AdditionalValueStep`. Would it be better to change those classes to extend from `ThreadResourceAbuseAdditionalTaintStep` and keep everything consistent? I wondered if the original author's intent was to eventually have those other AdditionalValueSteps as part of the core Java library instead of something specific to this query.

> The proper way of doing this would be using taint tracking and also adding arithmetic operations as default steps to the taint tracking library, since AFAIK they aren't currently included.
A response to issue #4845 mentioned that the CodeQL Java library deliberately doesn't include arithmetic operations as taint-propagating steps by default. I didn't want to shake the boat too much on my first PR 😎